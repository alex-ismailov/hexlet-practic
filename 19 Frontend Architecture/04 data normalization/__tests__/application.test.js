import fs from 'fs';
import path from 'path';
import prettier from 'prettier';

import run from '../src/application.js';

const options = {
  parser: 'html',
  htmlWhitespaceSensitivity: 'ignore',
  tabWidth: 4,
};

const fixturesPath = path.join(__dirname, '__fixtures__');
const getTree = () => prettier.format(document.body.innerHTML, options);

beforeEach(() => {
  const initHtml = fs.readFileSync(path.join(fixturesPath, 'index.html')).toString();
  document.documentElement.innerHTML = initHtml;
  run();
});

test('application', async () => {
  expect(getTree()).toMatchSnapshot();

  const newTaskForm = document.querySelector('[data-container="new-task-form"]');
  const newTaskNameInput = newTaskForm.elements.name;
  newTaskNameInput.setAttribute('value', 'First Task in General');
  newTaskForm.dispatchEvent(new Event('submit'));
  expect(getTree()).toMatchSnapshot();

  newTaskNameInput.setAttribute('value', 'Second Task in General');
  newTaskForm.dispatchEvent(new Event('submit'));
  expect(getTree()).toMatchSnapshot();

  const newListForm = document.querySelector('[data-container="new-list-form"]');
  const newListNameInput = newListForm.elements.name;
  newListNameInput.setAttribute('value', 'Random');
  newListForm.dispatchEvent(new Event('submit'));
  expect(getTree()).toMatchSnapshot();

  const randomList = document.querySelector('[href="#random"]');
  randomList.click();
  expect(getTree()).toMatchSnapshot();

  newTaskNameInput.setAttribute('value', 'First Task in Random');
  newTaskForm.dispatchEvent(new Event('submit'));
  expect(getTree()).toMatchSnapshot();

  const generalList = document.querySelector('[href="#general"]');
  generalList.click();
  expect(getTree()).toMatchSnapshot();
});

test('application 2', async () => {
  expect(getTree()).toMatchSnapshot();
});
