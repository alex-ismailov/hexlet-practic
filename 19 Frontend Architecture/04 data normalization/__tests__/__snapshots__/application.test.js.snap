// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`application 1`] = `
"<div class=\\"container m-3\\">
    <div class=\\"row\\">
        <div class=\\"col-2\\">
            <h3>Lists</h3>
            <div data-container=\\"lists\\">
                <ul>
                    <li><b>General</b></li>
                </ul>
            </div>
            <form data-container=\\"new-list-form\\">
                <input type=\\"text\\" class=\\"form-control mb-2\\" name=\\"name\\" />
                <input
                    type=\\"submit\\"
                    class=\\"btn btn-primary btn-sm\\"
                    value=\\"Add\\"
                />
            </form>
        </div>
        <div class=\\"col-10\\">
            <h3>Tasks</h3>
            <form class=\\"form-inline\\" data-container=\\"new-task-form\\">
                <input type=\\"text\\" class=\\"form-control mr-2\\" name=\\"name\\" />
                <input type=\\"submit\\" class=\\"btn btn-primary\\" value=\\"Add\\" />
            </form>
            <div data-container=\\"tasks\\"></div>
        </div>
    </div>
</div>
"
`;

exports[`application 2`] = `
"<div class=\\"container m-3\\">
    <div class=\\"row\\">
        <div class=\\"col-2\\">
            <h3>Lists</h3>
            <div data-container=\\"lists\\">
                <ul>
                    <li><b>General</b></li>
                </ul>
            </div>
            <form data-container=\\"new-list-form\\">
                <input type=\\"text\\" class=\\"form-control mb-2\\" name=\\"name\\" />
                <input
                    type=\\"submit\\"
                    class=\\"btn btn-primary btn-sm\\"
                    value=\\"Add\\"
                />
            </form>
        </div>
        <div class=\\"col-10\\">
            <h3>Tasks</h3>
            <form class=\\"form-inline\\" data-container=\\"new-task-form\\">
                <input
                    type=\\"text\\"
                    class=\\"form-control mr-2\\"
                    name=\\"name\\"
                    value=\\"First Task in General\\"
                />
                <input type=\\"submit\\" class=\\"btn btn-primary\\" value=\\"Add\\" />
            </form>
            <div data-container=\\"tasks\\">
                <ul>
                    <li>First Task in General</li>
                </ul>
            </div>
        </div>
    </div>
</div>
"
`;

exports[`application 2 1`] = `
"<div class=\\"container m-3\\">
    <div class=\\"row\\">
        <div class=\\"col-2\\">
            <h3>Lists</h3>
            <div data-container=\\"lists\\">
                <ul>
                    <li><b>General</b></li>
                </ul>
            </div>
            <form data-container=\\"new-list-form\\">
                <input type=\\"text\\" class=\\"form-control mb-2\\" name=\\"name\\" />
                <input
                    type=\\"submit\\"
                    class=\\"btn btn-primary btn-sm\\"
                    value=\\"Add\\"
                />
            </form>
        </div>
        <div class=\\"col-10\\">
            <h3>Tasks</h3>
            <form class=\\"form-inline\\" data-container=\\"new-task-form\\">
                <input type=\\"text\\" class=\\"form-control mr-2\\" name=\\"name\\" />
                <input type=\\"submit\\" class=\\"btn btn-primary\\" value=\\"Add\\" />
            </form>
            <div data-container=\\"tasks\\"></div>
        </div>
    </div>
</div>
"
`;

exports[`application 3`] = `
"<div class=\\"container m-3\\">
    <div class=\\"row\\">
        <div class=\\"col-2\\">
            <h3>Lists</h3>
            <div data-container=\\"lists\\">
                <ul>
                    <li><b>General</b></li>
                </ul>
            </div>
            <form data-container=\\"new-list-form\\">
                <input type=\\"text\\" class=\\"form-control mb-2\\" name=\\"name\\" />
                <input
                    type=\\"submit\\"
                    class=\\"btn btn-primary btn-sm\\"
                    value=\\"Add\\"
                />
            </form>
        </div>
        <div class=\\"col-10\\">
            <h3>Tasks</h3>
            <form class=\\"form-inline\\" data-container=\\"new-task-form\\">
                <input
                    type=\\"text\\"
                    class=\\"form-control mr-2\\"
                    name=\\"name\\"
                    value=\\"Second Task in General\\"
                />
                <input type=\\"submit\\" class=\\"btn btn-primary\\" value=\\"Add\\" />
            </form>
            <div data-container=\\"tasks\\">
                <ul>
                    <li>First Task in General</li>
                    <li>Second Task in General</li>
                </ul>
            </div>
        </div>
    </div>
</div>
"
`;

exports[`application 4`] = `
"<div class=\\"container m-3\\">
    <div class=\\"row\\">
        <div class=\\"col-2\\">
            <h3>Lists</h3>
            <div data-container=\\"lists\\">
                <ul>
                    <li><b>General</b></li>
                    <li><a href=\\"#random\\">Random</a></li>
                </ul>
            </div>
            <form data-container=\\"new-list-form\\">
                <input
                    type=\\"text\\"
                    class=\\"form-control mb-2\\"
                    name=\\"name\\"
                    value=\\"Random\\"
                />
                <input
                    type=\\"submit\\"
                    class=\\"btn btn-primary btn-sm\\"
                    value=\\"Add\\"
                />
            </form>
        </div>
        <div class=\\"col-10\\">
            <h3>Tasks</h3>
            <form class=\\"form-inline\\" data-container=\\"new-task-form\\">
                <input
                    type=\\"text\\"
                    class=\\"form-control mr-2\\"
                    name=\\"name\\"
                    value=\\"Second Task in General\\"
                />
                <input type=\\"submit\\" class=\\"btn btn-primary\\" value=\\"Add\\" />
            </form>
            <div data-container=\\"tasks\\">
                <ul>
                    <li>First Task in General</li>
                    <li>Second Task in General</li>
                </ul>
            </div>
        </div>
    </div>
</div>
"
`;

exports[`application 5`] = `
"<div class=\\"container m-3\\">
    <div class=\\"row\\">
        <div class=\\"col-2\\">
            <h3>Lists</h3>
            <div data-container=\\"lists\\">
                <ul>
                    <li><a href=\\"#general\\">General</a></li>
                    <li><b>Random</b></li>
                </ul>
            </div>
            <form data-container=\\"new-list-form\\">
                <input
                    type=\\"text\\"
                    class=\\"form-control mb-2\\"
                    name=\\"name\\"
                    value=\\"Random\\"
                />
                <input
                    type=\\"submit\\"
                    class=\\"btn btn-primary btn-sm\\"
                    value=\\"Add\\"
                />
            </form>
        </div>
        <div class=\\"col-10\\">
            <h3>Tasks</h3>
            <form class=\\"form-inline\\" data-container=\\"new-task-form\\">
                <input
                    type=\\"text\\"
                    class=\\"form-control mr-2\\"
                    name=\\"name\\"
                    value=\\"Second Task in General\\"
                />
                <input type=\\"submit\\" class=\\"btn btn-primary\\" value=\\"Add\\" />
            </form>
            <div data-container=\\"tasks\\"></div>
        </div>
    </div>
</div>
"
`;

exports[`application 6`] = `
"<div class=\\"container m-3\\">
    <div class=\\"row\\">
        <div class=\\"col-2\\">
            <h3>Lists</h3>
            <div data-container=\\"lists\\">
                <ul>
                    <li><a href=\\"#general\\">General</a></li>
                    <li><b>Random</b></li>
                </ul>
            </div>
            <form data-container=\\"new-list-form\\">
                <input
                    type=\\"text\\"
                    class=\\"form-control mb-2\\"
                    name=\\"name\\"
                    value=\\"Random\\"
                />
                <input
                    type=\\"submit\\"
                    class=\\"btn btn-primary btn-sm\\"
                    value=\\"Add\\"
                />
            </form>
        </div>
        <div class=\\"col-10\\">
            <h3>Tasks</h3>
            <form class=\\"form-inline\\" data-container=\\"new-task-form\\">
                <input
                    type=\\"text\\"
                    class=\\"form-control mr-2\\"
                    name=\\"name\\"
                    value=\\"First Task in Random\\"
                />
                <input type=\\"submit\\" class=\\"btn btn-primary\\" value=\\"Add\\" />
            </form>
            <div data-container=\\"tasks\\">
                <ul>
                    <li>First Task in Random</li>
                </ul>
            </div>
        </div>
    </div>
</div>
"
`;

exports[`application 7`] = `
"<div class=\\"container m-3\\">
    <div class=\\"row\\">
        <div class=\\"col-2\\">
            <h3>Lists</h3>
            <div data-container=\\"lists\\">
                <ul>
                    <li><b>General</b></li>
                    <li><a href=\\"#random\\">Random</a></li>
                </ul>
            </div>
            <form data-container=\\"new-list-form\\">
                <input
                    type=\\"text\\"
                    class=\\"form-control mb-2\\"
                    name=\\"name\\"
                    value=\\"Random\\"
                />
                <input
                    type=\\"submit\\"
                    class=\\"btn btn-primary btn-sm\\"
                    value=\\"Add\\"
                />
            </form>
        </div>
        <div class=\\"col-10\\">
            <h3>Tasks</h3>
            <form class=\\"form-inline\\" data-container=\\"new-task-form\\">
                <input
                    type=\\"text\\"
                    class=\\"form-control mr-2\\"
                    name=\\"name\\"
                    value=\\"First Task in Random\\"
                />
                <input type=\\"submit\\" class=\\"btn btn-primary\\" value=\\"Add\\" />
            </form>
            <div data-container=\\"tasks\\">
                <ul>
                    <li>First Task in General</li>
                    <li>Second Task in General</li>
                </ul>
            </div>
        </div>
    </div>
</div>
"
`;
